src/lib.rs

```diff
@@ -0,0 +1,1211 @@
+		let _background_processor = BackgroundProcessor::start(
+			Arc::clone(&self.persister),
+			Arc::clone(&event_handler),
+			Arc::clone(&self.chain_monitor),
+			Arc::clone(&self.channel_manager),
+			BPGossipSync::p2p(Arc::clone(&self.gossip_sync)),
+			Arc::clone(&self.peer_manager),
+			Arc::clone(&self.logger),
+			Some(Arc::clone(&self.scorer)),
+		);
+
+		// TODO: frequently check back on background_processor if there was an error
+		Ok(Runtime { tokio_runtime, _background_processor, stop_networking, stop_wallet_sync })
+	}
+
+	/// Blocks until the next event is available.
+	///
+	/// Note: this will always return the same event until handling is confirmed via [`Node::event_handled`].
+	pub fn next_event(&self) -> Event {
+		self.event_queue.next_event()
+	}
+
+	/// Confirm the last retrieved event handled.
+	pub fn event_handled(&self) {
+		self.event_queue.event_handled().unwrap();
+	}
+
+	/// Returns our own node id
+	pub fn node_id(&self) -> PublicKey {
+		self.channel_manager.get_our_node_id()
+	}
+
+	/// Returns our own listening address and port.
+	pub fn listening_address(&self) -> Option<String> {
+		self.config.listening_address.clone()
+	}
+
+	/// Retrieve a new on-chain/funding address.
+	pub fn new_funding_address(&self) -> Result<bitcoin::Address, Error> {
+		let funding_address = self.wallet.get_new_address()?;
+		log_info!(self.logger, "Generated new funding address: {}", funding_address);
+		Ok(funding_address)
+	}
+
+	/// Retrieve the current on-chain balance.
+	pub fn on_chain_balance(&self) -> Result<bdk::Balance, Error> {
+		self.wallet.get_balance()
+	}
+
+	/// Retrieve a list of known channels.
+	pub fn list_channels(&self) -> Vec<ChannelDetails> {
+		self.channel_manager.list_channels()
+	}
+
+	/// Connect to a node and open a new channel. Disconnects and re-connects are handled automatically
+	///
+	/// Returns a temporary channel id
+	pub fn connect_open_channel(
+		&self, node_pubkey_and_address: &str, channel_amount_sats: u64, announce_channel: bool,
+	) -> Result<(), Error> {
+		let runtime_lock = self.running.read().unwrap();
+		if runtime_lock.is_none() {
+			return Err(Error::NotRunning);
+		}
+
+		let runtime = runtime_lock.as_ref().unwrap();
+
+		let cur_balance = self.wallet.get_balance()?;
+		if cur_balance.get_spendable() < channel_amount_sats {
+			log_error!(self.logger, "Unable to create channel due to insufficient funds.");
+			return Err(Error::InsufficientFunds);
+		}
+
+		let peer_info = PeerInfo::try_from(node_pubkey_and_address.to_string())?;
+
+		let con_peer_info = peer_info.clone();
+		let con_success = Arc::new(AtomicBool::new(false));
+		let con_success_cloned = Arc::clone(&con_success);
+		let con_logger = Arc::clone(&self.logger);
+		let con_pm = Arc::clone(&self.peer_manager);
+
+		tokio::task::block_in_place(move || {
+			runtime.tokio_runtime.block_on(async move {
+				let res = connect_peer_if_necessary(
+					con_peer_info.pubkey,
+					con_peer_info.address,
```

Comment:
Could consider copy/cloning these fields here and below instead of the whole struct. That way you aren't cloning the whole struct twice if it happens to grown in number of fields.